lib/cursor-position-view.coffee

```diff
@@ -27,18 +29,34 @@ class CursorPositionView extends HTMLElement
   subscribeToConfig: ->
     @configSubscription?.dispose()
     @configSubscription = atom.config.observe 'status-bar.cursorPositionFormat', (value) =>
-      @formatString = value ? '%L:%C'
+      @statusFormat = value ? '%L:%C'
       @updatePosition()
 
   getActiveTextEditor: ->
     atom.workspace.getActiveTextEditor()
 
   updatePosition: ->
-    if position = @getActiveTextEditor()?.getCursorBufferPosition()
-      @row = position.row + 1
-      @column = position.column + 1
-      @textContent = @formatString.replace('%L', @row).replace('%C', @column)
+    if editor = @getActiveTextEditor()
+      screenpos = editor.getCursorScreenPosition()
+      bufpos = editor.getCursorBufferPosition()
+      @row = bufpos.row + 1
+      @column = screenpos.column + 1
+      @lineCount = editor.getLineCount()
+      @length = editor.getText().length
+      @offset = editor.getTextInBufferRange([[0, 1], bufpos]).length
+      @percent = Math.round(100 * @offset / @length)
+      @textContent = @formatString(@statusFormat)
```

Comment:
I'd rather not pay for these calculations on every cursor movement if I'm not displaying the information. I'm pretty sure that the row and column locations are fast, but is calculating the length of a 500MB file fast as well?